stellarphot/core.py

```diff
@@ -171,9 +171,14 @@ def _update_passbands(self):
         # in case any of the new names are longer than the longest of the old names.
         # If that happens, astropy by default just truncates the names.
         new_filter_name = [
-            self._passband_map[orig_pb] if orig_pb in self._passband_map else orig_pb
+            (
+                self._passband_map[orig_pb]
+                if orig_pb in self._passband_map.keys()
```

Comment:
Isn't the default iterator for a dictionary keys?  Oh, I see, you defined a method keys for the PassbandModel that returns the keys.  Later I do see where you perform `self._passband_map = passband_map.model_copy()` to set `self._passband_map`... but `passband_map` is a `dict` according to the docstrings for`CatalogData` and `PhotometryData` . Does this need to get changed to `PassbandMap` object?

Did you place this function the `BaseEnhancedTable` so it is available to be inherited by `CatalogData` and `PhotometryData`?  I ask because the passband mapping is not used in the `BaseEnhancedTable`.

Finally, what is `self["passband"]`?   I can't find where `self["passband"]` is even defined in `BaseEnhancedTable`?

Reply:
Taking a look in a moment -- the intent was to have every place use a `PassbandMap` object but to make that object act sort of like a dict.

Reply:
Also, you are write that the default iterator is over keys, but I wasn't sure which dunder method I needed to use to make that happen. Can dig it up in _Fluent Python_ probably.

Reply:
> Did you place this function the BaseEnhancedTable so it is available to be inherited by CatalogData and PhotometryData? I ask because the passband mapping is not used in the BaseEnhancedTable.

I think `_update_passbands` has been in `BaseEnhancedTable` for a while: https://github.com/feder-observatory/stellarphot/blob/main/stellarphot/core.py#L166

I do think you are correct about the reason it is in `BaseEnhancedTable`

Reply:
> Finally, what is self["passband"]?

Took me a sec to remember, it is one of the columns in the table.